src/itemdrop.zig

```diff
@@ -537,6 +573,16 @@ pub const ItemDisplayManager = struct { // MARK: ItemDisplayManager
 		const n2: Vec3f = @as(Vec3f, @splat(1)) + damping*@as(Vec3f, @splat(dt));
 		cameraFollowVel = n1/(n2*n2);
 
+		targetSwingProgress *= @min(3*currentSwingTime, 1);
+		if(!isSwinging) {
+			swing *= std.math.pow(f32, 0.05, dt);
+		} else {
+			swing = targetSwingProgress;
+		}
+		if(std.math.isNan(swing)) {
```

Comment:
If this really is something that can realistically happen, then you should be rethinking your approach.